zanata-war/src/main/java/org/zanata/action/LanguageJoinAction.java

```diff
@@ -75,54 +85,91 @@
     @In
     private Messages msgs;
 
-    @Getter
     @Setter
-    private boolean requestAsTranslator;
-
     @Getter
-    @Setter
-    private boolean requestAsReviewer;
+    private String language;
+
+    private HLocale locale;
 
     @Getter
     @Setter
-    private boolean requestAsCoordinator;
+    @Size(max = 1000)
+    private String message;
 
     @Setter
     @Getter
-    private String language;
+    private String declineMessage;
 
-    private HLocale locale;
+    @In
+    private RequestService requestServiceImpl;
 
-    @Getter
-    @Setter
-    private String message;
+    @In
+    private ZanataIdentity identity;
+
+    @In
+    private LanguageTeamService languageTeamServiceImpl;
 
     @In(value = ZanataJpaIdentityStore.AUTHENTICATED_USER, required = false)
     private HAccount authenticatedAccount;
 
-    public boolean hasSelectedRole() {
-        return requestAsTranslator || requestAsReviewer || requestAsCoordinator;
+    public String getMyRequestedRole() {
+        LanguageRequest request =
+            requestServiceImpl.getPendingLanguageRequests(authenticatedAccount,
+                getLocale().getLocaleId());
+        return getRequestedRole(request);
     }
 
-    public void bindRole(String role, boolean checked) {
-        if (role.equals("translator")) {
-            requestAsTranslator = checked;
-        } else if (role.equals("reviewer")) {
-            requestAsReviewer = checked;
-        } else if (role.equals("coordinator")) {
-            requestAsCoordinator = checked;
-        }
+    public String getRequestedRole(LanguageRequest request) {
```

Comment:
- [ ] SF: looks like this should be getRequestedRoles or getLocalisedRequestedRoles, and/or it needs a javadoc